### PR TITLE
[RAPTOR-5302] contains missing validation

### PR DESCRIPTION
**Pull request by scottp-ml**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Add support for the contains_missing validation field.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 1]
```diff
@@ -290,6 +324,56 @@ def validate(self, dataframe):
         return errors
 
 
+class InputContainsMissing(object):
```
Comment (4dahalibut) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:327]: Should some of this logic be extracted to a super?

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 2]
```diff
@@ -7,12 +7,38 @@
 import pandas as pd
 
 
+class Conditions:
```
Comment (4dahalibut) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:10]: Wondering if these should have more descriptive names.

Comment (scottp-ml) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:10]: I added descriptions.  For this I like them matching the field names in the specificaiton

**Review by 4dahalibut — APPROVED** [review]

**Comment by eric-s-s** [comment]
do tests for `custom_model_runner.datarobot_drum.drum.typeschema_validation.revalidate_typeschema` need to be updated to test the `get_yaml_validator`?

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py (hunk 2 above)
Comment (eric-s-s) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:15]: can/should these be actual Enums?

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 3]
```diff
@@ -12,21 +12,72 @@
...
+class DataTypes(BaseValidator):
     """Validation related to data types.  This is common between input and output."""
 
     FIELD = "data_types"
-    VALUES = ["NUM", "TXT", "CAT", "IMG", "DATE"]
-    CONDITIONS = ["EQUALS", "IN", "NOT_EQUALS", "NOT_IN"]
+    VALUES = [Values.NUM, Values.TXT, Values.CAT, Values.IMG, Values.DATE]
+    CONDITIONS = [Conditions.EQUALS, Conditions.IN, Conditions.NOT_EQUALS, Conditions.NOT_IN]
     _TYPES = Enum(VALUES)
     _DTYPE_MAPPING = {}
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:75]: may as well delete this

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py [hunk 4]
```diff
@@ -12,21 +12,72 @@
...
+    NEVER = "NEVER"
+    DYNAMIC = "DYNAMIC"
+    ALWAYS = "ALWAYS"
+    IDENTITY = "IDENTITY"
+
+
+class BaseValidator(object):
+    FIELD = None
+    CONDITIONS = None
+    VALUE = None
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/typeschema_validation.py:48]: VALUES

File: custom_model_runner/datarobot_drum/drum/typeschema_validation.py (hunk 2 above)
Comment (eric-s-s) [reply custom_model_runner/datarobot_drum/drum/typeschema_validation.py:15]: short answer, they _could_ but it might not be worth the time

**Review by eric-s-s — APPROVED** [review]
There are a couple of nits.  The one big ask I have is that a test is made for a yaml that uses `field: contains_missing`.  

:+1:

**Pull request merged**